WooCommerce/src/main/kotlin/com/woocommerce/android/tracker/TrackerRepository.kt

```diff
@@ -0,0 +1,28 @@
+package com.woocommerce.android.tracker
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.longPreferencesKey
+import com.woocommerce.android.datastore.DataStoreQualifier
+import com.woocommerce.android.datastore.DataStoreType
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import org.wordpress.android.fluxc.model.SiteModel
+import javax.inject.Inject
+
+class TrackerRepository @Inject constructor(
+    @DataStoreQualifier(DataStoreType.TRACKER) private val dataStore: DataStore<Preferences>
+) {
+    fun observeLastSendingDate(site: SiteModel): Flow<Long> {
+        return dataStore.data.map { preferences ->
+            preferences[longPreferencesKey(site.id.toString())] ?: 0
+        }
+    }
+
+    suspend fun updateLastSendingDate(site: SiteModel, lastUpdateMillis: Long) {
+        dataStore.edit { trackerPreferences ->
+            trackerPreferences[longPreferencesKey(site.id.toString())] = lastUpdateMillis
+        }
+    }
```

Comment:
I wonder if this is a good strategy or rather I should make it not `suspend` and provide dispatcher directly. I mean:

```suggestion
    fun updateLastSendingDate(site: SiteModel, lastUpdateMillis: Long) {
        CoroutineScope(coroutineDispatchers.io).launch {
            dataStore.edit { trackerPreferences ->
                trackerPreferences[longPreferencesKey(site.id.toString())] = lastUpdateMillis
            }
        }
    }
```

Reply:
Well, at least from the perspective of how we've been managing launch calls, I think it's OK to keep the `coroutineDispatchers` injection at the `ViewModel` (Presenter for this case) and call all suspend functions starting from there. Do you see a disadvantage of keeping doing this?

Reply:
Thanks! My concern was about delegating responsibility about running code on a correct context (and therefore dispatcher) directly to repository. We do this e.g. in `<*>Store` classes with `CoroutineEngine` if I'm not mistaken.

But now when code has been moved into `AppInitializer` and is executed in `@AppCoroutineScope` (aka `GlobalScope`) I think we're good (calls are executed on `Thread[DefaultDispatcher-worker-4,5,main]`).